ext/afform/core/Civi/Api4/Action/CustomGroup/GetAfforms.php

```diff
@@ -0,0 +1,375 @@
+      'description' => '',
+      'is_public' => FALSE,
+      'permission' => ['access CiviCRM'],
+      'entity_type' => $item['extends'],
+      'icon' => $item['icon'],
+    ];
+    if ($item['is_multiple']) {
+      $afform['join_entity'] = 'Custom_' . $item['name'];
+    }
+    if ($this->getLayout) {
+      $afform['layout'] = \CRM_Core_Smarty::singleton()->fetchWith(
+        'afform/customGroups/afblock.tpl',
+        ['group' => $item]
+      );
+    }
+    return $afform;
+  }
+
+  private function generateUpdateForm($item): array {
+    $afform = [
+      'name' => 'afformUpdateCustom_' . $item['name'],
+      'type' => 'form',
+      'title' => E::ts('Update %1', [1 => $item['title']]),
+      'description' => '',
+      'is_public' => FALSE,
+      // NOTE: we will use RBAC for entities to ensure
+      // this form does not allow folks who shouldn't
+      // to edit contacts
+      'permission' => ['access CiviCRM'],
+      'server_route' => 'civicrm/af/custom/' . $item['name'] . '/update',
+      'icon' => $item['icon'],
+    ];
+    if ($this->getLayout) {
+
+      // form entity depends on whether this is a multirecord custom group
+      $formEntity = $item['is_multiple'] ?
+        [
+          'type' => 'Custom_' . $item['name'],
+          'name' => 'Record',
+          'label' => $item['extends'] . ' ' . $item['title'],
+          'parent_field' => 'entity_id',
+          'parent_field_defn' => [
+            'input_type' => 'Hidden',
+            'label' => FALSE,
+          ],
+        ] :
+        [
+          'type' => $item['extends'],
+          'name' => $item['extends'] . '1',
+          'label' => $item['extends'],
+          'parent_field' => 'id',
+          'parent_field_defn' => [
+            'input_type' => 'Hidden',
+            'label' => FALSE,
+          ],
+        ];
+
+      $afform['layout'] = \CRM_Core_Smarty::singleton()->fetchWith(
+        'afform/customGroups/afform.tpl',
+        [
+          'formEntity' => $formEntity,
+          'formActions' => [
+            'create' => FALSE,
+            'update' => TRUE,
+          ],
+          'urlAutofill' => TRUE,
+          'blockDirective' => _afform_angular_module_name('afblockCustom_' . $item['name'], 'dash'),
+        ]
+      );
+    }
+    return $afform;
+  }
+
+  private function generateCreateForm($item): array {
+    $afform = [
+      'name' => 'afformCreateCustom_' . $item['name'],
+      'type' => 'form',
+      'title' => E::ts('Add %1', [1 => $item['title']]),
+      'description' => '',
+      'is_public' => FALSE,
+      // NOTE: we will use RBAC for entities to ensure
+      // this form does not allow folks who shouldn't
+      // to edit contacts
+      'permission' => ['access CiviCRM'],
+      'server_route' => 'civicrm/af/custom/' . $item['name'] . '/create',
+      'icon' => $item['icon'],
+    ];
+    if ($this->getLayout) {
+      $formEntity = [
+        'type' => 'Custom_' . $item['name'],
+        'name' => 'Record',
```

Comment:
Capitalizing "Record" leads to an odd uppercase letter in the paths.

Reply:
Yeh I wasn't sure about this. 

It's a halfway house between:
- the conventional afform entity name ( ie `Custom_PlayingCard1`) -- but that's annoying that you need to use the entity name in the path and the param
- just `id` which I think would be the sane name for the url param, but not so much for the entity on the form

Reply:
Ok that makes sense.